docs/directives/need.rst

```diff
@@ -38,12 +38,127 @@ but you must set a title as an argument (i.e. if you do not specify :ref:`needs_
+  With multiple variant definitions, we set the first matching variant as the *need option's* value.
+* When you set a *need option* to multiple variant definitions, you can specify the last definition as
+  a default "variant-free" option which we can use if no variant definition matches. |br|
+  Example; In this multi-variant definitions, ``[status in tags]:added; var_a:changed; unknown``, *unknown* will be used
+  if none of the other variant definitions are True.
+* If you prefer your variant definitions to use rules instead of keys, then you should put your filter string
+  inside square brackets like this: ``['name' in tags]:assigned``.
+* For multi-variant definitions, you can mix both rule and variant-named options like this:
+  ``[author["test"][0:4] == 'me']:Me, var_a:Variant A, Unknown``
+
+To implement variants options, you must configure the following in your ``conf.py`` file:
+
+* :ref:`needs_variants`
+* :ref:`needs_variant_options`
+* :ref:`needs_variant_data`
+
+There are various use cases for variants options support.
+
+Use Case 1
+~~~~~~~~~~
+
+In this example, you set the :ref:`needs_variants` configuration that comprises pre-defined variants assigned to
+"filter strings".
+You can then use the keys in your ``needs_variants`` as references when defining variants for a *need option*.
+
+|ex|
+
+In ``conf.py``:
+
+.. code-block:: python
+
+   needs_variants = {
+     "var_a": "'var_a' in sphinx_tags"  # filter_string
+     "var_b": "assignee == 'me'"
+   }
+
+In your ``.rst`` file:
+
+.. code-block:: rst
+
+   .. req:: Example
+      :id: VA_001
+      :status: var_a:open, var_b:closed, unknown
+
+From the above example, if a *need option* has variants defined, then we get the filter string
+from our ``needs_variants`` configuration and evaluate it.
+If a variant definition is true, then we set the *need option* to the value of the variant definition.
+
+Use Case 2
+~~~~~~~~~~
+
+In this example, you can use the filter string directly in the *need option's* variant definition.
+
+|ex|
+
+In your ``.rst`` file:
+
+.. code-block:: rst
+
+   .. req:: Example
+      :id: VA_002
+      :status: ['var_a' in tags]:open, [assignee == 'me']:closed, unknown
+
+From the above example, we evaluate the filter string in our variant definition without referring to :ref:`needs_variants`.
+If a variant definition is true, then we set the *need option* to the value of the variant definition.
+
+Below is an implementation of variants for need options:
+
+|ex|
+
+.. code-block:: rst
+
+   .. req:: Variant options
+      :id: VA_003
+      :status: ['variants' in tags and not collapse]:enabled, disabled
+      :tags: variants;support
+      :collapse: true
+
+      Variants for need options in action
+
+|out|
+
+.. req:: Variant options
+   :id: VA_003
+   :status: ['variants' in tags and not collapse]:enabled, disabled
+   :tags: variants;support
+   :collapse: true
+
+   Variants for need options in action
+
```

Comment:
Use Case 3 missing :)

Using [Sphinx-Tags](https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-t)

If docs get built via `sphinx-build -b html -t tag_a . _build`.
```rst
  .. req:: Example
      :id: VA_001
      :status: tag_a:open, tag_b:closed, unknown
```
"open" is set for status.

Reply:
I will read about it. Thanks.

Reply:
Included in update.